### PR TITLE
Cleanup and Test Improvements

### PR DESCRIPTION
**Pull request by jwair**
Improved tests - added missing asserts, removed overriding builtin 'delete', unified naming
Removed redundant type declarations and casting
Removed conflicts with package names
Improved naming and comments
Unified usage of 'require' test package instead of a mix of require and assert

### CODE REVIEW

**Comment by jwair** [comment]
Working on the e2e test failures ...

File: e2e/command_test.go [hunk 1]
```diff
@@ -244,7 +245,8 @@ func TestE2E_ReenableTaskTriggers(t *testing.T) {
 
 	cts, stop := api.StartCTS(t, configPath)
 	t.Cleanup(func() {
-		cleanup()
+		err := cleanup()
+		require.NoError(t, err)
```
Comment (mkam) [inline e2e/command_test.go:249]: I'm not so sure we should be failing on the cleanup like this. If there is an issue with deleting a temporary directory, then we'd have to rerun the tests even though the behavior of CTS was passing.

Comment (jwair) [reply e2e/command_test.go]: Addressed in https://github.com/hashicorp/consul-terraform-sync/pull/653/commits/40cb2fc4f6637d896e608977841e3c4e022645ac

**Review by mkam — APPROVED** [review]
Changes look good to me! Thanks for all the improvements!

**Pull request merged**